interface/product_registration/product_registration_controller.php

```diff
@@ -18,14 +18,64 @@
  * @link    http://www.open-emr.org
  */
 
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
+$ignoreAuth=true;
 include_once("../globals.php");
+require_once($GLOBALS['fileroot'] . "/interface/product_registration/product_registration_service.php");
+require_once($GLOBALS['fileroot'] . "/interface/main/exceptions/invalid_email_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/main/exceptions/local_product_registration_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/main/exceptions/remote_product_registration_exception.php");
 
-// TODO: Since this is a simple endpoint, map inbound HTTP GET to "get" and POST to "post" actions
+class ProductRegistrationController {
+    private $productRegistrationService;
 
-function get() {
-    //call to ProductRegistrationService.getProductStatus()
-}
+    public function __construct($httpRequestMethod) {
+        $this->productRegistrationService = new ProductRegistrationService();
+
+        switch ($httpRequestMethod) {
+            case 'POST':
+                $this->post();
+                break;
+            case 'GET':
+                $this->get();
+                break;
+        }
+    }
+
+    public function get() {
+        $statusPayload = new stdClass();
+        $statusPayload->status = $this->productRegistrationService->getProductStatus();
+
+        header('Content-Type: application/json');
+        http_response_code(200);
+        echo json_encode($statusPayload);
+    }
 
-function post() {
-    //call to ProductRegistrationService.registerProduct or optOutOfProductRegistration
+    public function post() {
+        $registrationPayload = new stdClass();
+        $status = null;
+
+        try {
+            $registrationId = $this->productRegistrationService->registerProduct($_POST["email"]);
+            $registrationPayload->id = $registrationId;
+            $status = 201;
+        } catch (InvalidEmailException $emailException) {
+            $registrationPayload->error = $emailException->errorMessage();
+            $status = 400;
+        } catch (LocalProductRegistrationException $localRegistrationException) {
+            $registrationPayload->error = $localRegistrationException->errorMessage();
+            $status = 500;
+        } catch (RemoteProductRegistrationException $remoteRegistrationException) {
+            $registrationPayload->error = $remoteRegistrationException->errorMessage();
+            $status = 500;
+        }
+
+        header('Content-Type: application/json');
+        http_response_code($status);
+        echo json_encode($registrationPayload);
```

Comment:
Note I've added more calls to this in https://github.com/openemr/openemr/pull/257/commits/ddea008fb85db258bb51d5aa57f24e242eb622cc to disable caching...